### DISCUSSION TITLE
LDAP authentication

### DISCUSSION THREAD

**Discussion by elvinu** [jonaswinkler/paperless-ng#654] [Feature Requests]
Hi there,

In paperless was a feature request for [LDAP backend auth](https://github.com/the-paperless-project/paperless/issues/239). I was wondering if can be implemented in paperless-ng. 

There was a solution from [nebulade - Cludron](https://github.com/the-paperless-project/paperless/issues/239#issuecomment-338837279) if someone can look over it and see if can be integrated in paperless-ng it would be awesome.

Thank you.

**Comment by jonaswinkler** [comment]
Hello!

I don't exactly need LDAP, so I won't implement that. I also don't have a setup to test that.

However, it sounds like a really good thing to have and I'd be open to PRs for that. The [Django LDAP plugin](https://github.com/django-auth-ldap/django-auth-ldap)  is pretty active, and it's mostly a configuration issue.

If you or someone else wants to take care of that:

* Make a PR, using the current master as a base, to a new feature branch (lets say `feature-ldap-auth`).
* These get processed by the CI pipeline as well, so you'll be able to test the docker image before we're rolling this into a release.
* If it works, I'll have that merged into a release.

**Reply by marcusquinn** [reply]
@jonaswinkler Wondering if you might find the joys of LDAP across self-hosted webapps if you give Cloudron a try? I'm happy to recommend it's worth it if you like self-hosting and making life easier :)

**Comment by DoctorOgg** [comment]
i would love to see LDAP auth, I do run it for home and work Specifically OpenLdap with Samba, would be really nice to have parity, esp for access to the consume folder via SMB

**Comment by tweak19** [comment]
There are any Updates about using LDAP-Auth with paperless-ng?
I already tried to get it working depending on the info from nebulade but didn't work.
Would be pretty awesome if LDAP Auth, works!

**Comment by dani** [comment]
I've made a PR to add LDAP auth support (https://github.com/jonaswinkler/paperless-ng/pull/1211) but would gladly take some help to debug the ansible test failure, and the issue with the requirements.txt file

**Comment by MAngel666** [comment]
+1
